pkg/apis/templating/templating.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templating
+
+import (
+	"bytes"
+	"text/template"
+
+	csiapi "github.com/jetstack/cert-manager-csi/pkg/apis/v1alpha1"
+)
+
+func RenderAttributeTemplates(attr map[string]string) (map[string]string, error) {
+	var err error
+
+	tmpls := &template.Template{}
+	if err = addAndParseTemplate(tmpls, csiapi.CommonNameKey); err != nil {
+		return nil, err
+	}
+	if err = addAndParseTemplate(tmpls, csiapi.DNSNamesKey); err != nil {
+		return nil, err
+	}
```

Comment:
Maybe add URI in this PR?

IP doesn't make sense until we can gather intelligence about the pods IP (maybe in a follow up PR).